src/components/activity/ActivityTableMenu.svelte

```diff
@@ -0,0 +1,163 @@
+<svelte:options immutable={true} />
+
+<script lang="ts">
+  import SearchIcon from '@nasa-jpl/stellar/icons/search.svg?component';
+  import type { ColDef, ColumnState } from 'ag-grid-community';
+  import { createEventDispatcher } from 'svelte';
+  import Input from '../form/Input.svelte';
+  import Menu from '../menus/Menu.svelte';
+  import MenuItem from '../menus/MenuItem.svelte';
+
+  export let columnDefs: ColDef[] = [];
+  export let columnStates: ColumnState[] = [];
+
+  interface ColumnMenuItem {
+    field: keyof Activity;
+    isHidden: boolean;
+    name: string;
+  }
+
+  const dispatch = createEventDispatcher();
+
+  let columnMenuItems: ColumnMenuItem[] = [];
+  let displayedColumnMenuItems: ColumnMenuItem[] = [];
+  let searchFilter: string = '';
+  let tableMenu: Menu;
+
+  $: columnMenuItems = columnDefs.map((derivedColumnDef: ColDef) => {
+    const columnState = columnStates.find((columnState: ColumnState) => columnState.colId === derivedColumnDef.field);
+
+    if (columnState) {
+      return {
+        field: derivedColumnDef.field as keyof Activity,
+        isHidden: columnState?.hide ?? derivedColumnDef.hide ?? false,
+        name: derivedColumnDef.headerName,
+      };
+    }
+
+    return {
+      field: derivedColumnDef.field as keyof Activity,
+      isHidden: true,
+      name: derivedColumnDef.headerName,
+    };
+  });
+
+  $: displayedColumnMenuItems = columnMenuItems.filter((columnMenuItem: ColumnMenuItem) => {
+    return new RegExp(searchFilter, 'i').test(columnMenuItem.field);
+  });
+
+  function onColumnToggleChange(column: ColumnMenuItem) {
+    dispatch('toggle-column', {
+      field: column.field,
+      isHidden: !column.isHidden,
+    });
+  }
+
+  function onHideAllColumns() {
+    dispatch('hide-all-columns');
+  }
+
+  function onShowAllColumns() {
+    dispatch('show-all-columns');
+  }
+
+  function onSearchFilterChange(event: Event) {
+    const input = event.target as HTMLInputElement;
+    searchFilter = input.value;
+  }
+</script>
+
+<div class="grid-menu st-typography-medium" on:click|stopPropagation={() => tableMenu.toggle()}>
```

Comment:
Add `<!-- svelte-ignore a11y-click-events-have-key-events -->` to suppress warning